counterpartylib/lib/api.py

```diff
@@ -382,21 +384,34 @@ def run(self):
 
 class APIServer(threading.Thread):
     """Handle JSON-RPC API calls."""
-    def __init__(self):
+    def __init__(self, db=None):
+        self.db = db
         self.is_ready = False
+        self.shutdown_secret = binascii.hexlify(os.urandom(32)).decode('ascii')
         threading.Thread.__init__(self)
         self.stop_event = threading.Event()
 
+        self.app = None
+
     def stop(self):
+        util.api('shutdown', {'secret': self.shutdown_secret})
```

Comment:
I cant say. I think this may have come from another branch I depended upon that has not been merged into dev yet. @rubensayshi do you know more about this?